rotkehlchen/chain/ethereum/uniswap/uniswap.py

```diff
@@ -0,0 +1,367 @@
+    ) -> ProtocolBalance:
+        """Get the addresses' pools data querying the Uniswap subgraph
+        """
+        address_balances: AddressBalances = {address: [] for address in addresses}
+        known_assets: Set[EthereumToken] = set()
+        unknown_assets: Set[UnknownEthereumToken] = set()
+
+        addresses_lower = [address.lower() for address in addresses]
+        querystr = format_query_indentation(LIQUIDITY_POSITIONS_QUERY.format())
+        param_types = {
+            '$limit': 'Int!',
+            '$offset': 'Int!',
+            '$addresses': '[String!]',
+            '$balance': 'BigDecimal!',
+        }
+        param_values = {
+            'limit': GRAPH_QUERY_LIMIT,
+            'offset': 0,
+            'addresses': addresses_lower,
+            'balance': "0",
+        }
+        while True:
+            result = graph_query(
+                querystr=querystr,
+                param_types=param_types,
+                param_values=param_values,
+            )
+            result_data = result['liquidityPositions']
+
+            for lp in result_data:
+                user_address = to_checksum_address(lp['user']['id'])
+                user_lp_balance = FVal(lp['liquidityTokenBalance'])
+                lp_pair = lp['pair']
+                lp_address = to_checksum_address(lp_pair['id'])
+                lp_total_supply = FVal(lp_pair['totalSupply'])
+
+                # Insert LP tokens reserves within tokens dicts
+                token0 = lp_pair['token0']
+                token0['total_amount'] = lp_pair['reserve0']
+                token1 = lp_pair['token1']
+                token1['total_amount'] = lp_pair['reserve1']
+
+                liquidity_pool_assets = []
+
+                for token in token0, token1:
+                    # Get the token <EthereumToken> or <UnknownEthereumToken>
+                    asset = get_ethereum_token(
+                        symbol=token['symbol'],
+                        ethereum_address=to_checksum_address(token['id']),
+                        name=token['name'],
+                        decimals=int(token['decimals']),
+                    )
+
+                    # Classify the asset either as known or unknown
+                    if isinstance(asset, EthereumToken):
+                        known_assets.add(asset)
+                    elif isinstance(asset, UnknownEthereumToken):
+                        unknown_assets.add(asset)
+
+                    # Estimate the underlying asset total_amount
+                    asset_total_amount = FVal(token['total_amount'])
+                    user_asset_balance = (
+                        user_lp_balance / lp_total_supply * asset_total_amount
+                    )
+
+                    liquidity_pool_asset = LiquidityPoolAsset(
+                        asset=asset,
+                        total_amount=asset_total_amount,
+                        user_balance=Balance(amount=user_asset_balance),
```

Comment:
Both here and below in line 158 you are initializing the balance with a 0 usd value on purpose, right?

Reply:
yes